sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -171,8 +113,56 @@ private void setNotNullAt(int i) {
   }
 
   @Override
-  public void update(int ordinal, Object value) {
-    throw new UnsupportedOperationException();
+  public void update(int i, Object value) {
+    if (value == null) {
```

Comment:
can you add some comments inline on what is happening? It is pretty hard to follow for this function.